onebusaway-android/src/amazon/java/org/onebusaway/android/map/googlemapsv2/BaseMapFragment.java

```diff
@@ -357,6 +360,23 @@ protected void showDialog(int id) {
         MapDialogFragment.newInstance(id, this).show(getFragmentManager(), MapDialogFragment.TAG);
     }
 
+    private void setLocationByZip(final String zip) {
+        Runnable r = new Runnable() {
+            @Override
+            public void run() {
+                Location l = LocationUtils.getLocationFromZip(getContext(), zip);
+                if (l != null) {
+                    mLocationHelper.onLocationChanged(l);
```

Comment:
We may need to check if `mLocationHelper == null` here as well - `LocationUtils.getLocationFromZip()` does a network call and could potentially be a long-running method, and the user may have backgrounded or killer the Fragment/Activity in the mean time.  Please do some testing where you enter the zip and then immediately kill the app to make sure it doesn't crash.